File: PjblRa4/SorteadorPalavras.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Random;

public class SorteadorPalavras {
    private ArrayList<String> palavras;
    private Random random;

    public SorteadorPalavras() {
        palavras = new ArrayList<>();
        random = new Random();
    }

    public void adicionarPalavra(String palavra) {
        if (palavra != null && !palavra.isEmpty()) {
            palavras.add(palavra);
        }
    }

    public List<String> sortearPalavras(int quantidade) {
        if (quantidade <= 0 || quantidade > palavras.size()) {
            return Collections.emptyList(); 
        }

        List<String> copiaPalavras = new ArrayList<>(palavras);
        Collections.shuffle(copiaPalavras, random);
        return copiaPalavras.subList(0, quantidade);
    }

    public ArrayList<String> getPalavras() {
        return palavras;
    }
}

File: Ajuste/SorteadorPalavras.py
import random

class SorteadorPalavras:
    def __init__(self):
        self.palavras = []

    def adicionar_palavra(self, palavra):
        if palavra:
            self.palavras.append(palavra)

    def sortear_palavras(self, quantidade):
        if quantidade <= 0:
            return []

        quantidade = min(quantidade, len(self.palavras))

        copia_palavras = self.palavras[:]
        random.shuffle(copia_palavras)
        return copia_palavras[:quantidade]

    def get_palavras(self):
        return self.palavras
